File: logger/log_row.cpp
#include "log_row.h"
#include "log_message_type_enum.h"

#include <string>

namespace SausoonTech
{
	namespace Logger
	{
		LogRow::LogRow(const LogMessageTypeEnum* log_message_type, const std::string* message)
		{
			log_message_type_ = *log_message_type;
			message_ = *message;
		}

		LogMessageTypeEnum LogRow::get_log_message_type() const
		{
			return log_message_type_;
		}

		std::string LogRow::get_message() const
		{
			return message_;
		}
	}
}

File: SausoonTech.Logger/LogRow.cs
namespace SausoonTech.Logger
{
    using System;

    using log4net;
    using log4net.Core;

    internal class LogRow
    {
        internal Level Level { get; }
        internal string Message { get; }
        internal Exception Exception { get; }

        internal LogRow(Level level, string message, Exception exception)
        {
            this.Level = level;
            this.Message = message;
            this.Exception = exception;
        }
    }
}